### ISSUE TITLE
GetConnect PATCH method

### ISSUE THREAD

**Issue by Artel-Studio** [jonataslaw/getx#2757]
**Describe the bug**
Flutter web. Trying to make PATCH request thru Google Chrome, but getting error: "Access to XMLHttpRequest at '...' from origin '...' has been blocked by CORS policy: Method patch is not allowed by Access-Control-Allow-Methods in preflight response."

CORS is well configured for using PATCH method (Access-Control-Allow-Methods: *).

**Flutter Version:**
Flutter 3.7.7 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 2ad6cd72c0 (3 недели назад) • 2023-03-08 09:41:59 -0800
Engine • revision 1837b5be5f
Tools • Dart 2.19.4 • DevTools 2.20.1

**Getx Version:**
GetX 4.6.5

**Solution**
The only way to solve the problem was to write the word 'patch' in upper case ('PATCH') in
get_connect/http/src/http.dart (line 332).

As a temporary measure (until a package update comes out), I rewrote the method in my child class like this:

`class MyChildClass extends GetConnect {
@override
  Future<Response<T>> patch<T>(String url, dynamic body, {
    String? contentType,
    Map<String, String>? headers,
    Map<String, dynamic>? query,
    Decoder<T>? decoder,
    Progress? uploadProgress,
  }) {
    return super.request<T>(
      url,
      'PATCH',
      body: body,
      headers: headers,
      contentType: contentType,
      query: query,
      decoder: decoder,
      uploadProgress: uploadProgress,
    );
  }
}
`

**Comment by jonataslaw** [comment]
You need enable cors in your server

**Comment by TheFe91** [comment]
@jonataslaw I'm having the exact same issue. The funny fact is that the problem only occurs with the patch method (get, post, put, delete all work fine) in a browser environment (aka flutter web). I don't get why

**Comment by leonardtan13** [comment]
Same issue here. Seems like the package is using `patch` instead of `PATCH`, which is why @Artel-Studio 's fix to `PATCH` works. 

https://fetch.spec.whatwg.org/#methods 
https://stackoverflow.com/questions/73985866/the-mystery-of-patch-request-and-cors-headers

![image](https://github.com/jonataslaw/getx/assets/54533009/7df81bbd-95e1-43ca-a9ce-e77e90be3a00)
Here we can see that the request is a `patch` request, resulting in a CORS error as the server returns `PATCH` as one of the allowed methods

The strange thing is, this seems to not be an issue in Safari, probably due to how the different browsers implement CORS checking.

**Comment by inyong1** [comment]
> Same issue here. Seems like the package is using `patch` instead of `PATCH`, which is why @Artel-Studio 's fix to `PATCH` works.
> 
> https://fetch.spec.whatwg.org/#methods https://stackoverflow.com/questions/73985866/the-mystery-of-patch-request-and-cors-headers
> 
> ![image](https://private-user-images.githubusercontent.com/54533009/327043944-7df81bbd-95e1-43ca-a9ce-e77e90be3a00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1NjA0NTQsIm5iZiI6MTcxNDU2MDE1NCwicGF0aCI6Ii81NDUzMzAwOS8zMjcwNDM5NDQtN2RmODFiYmQtOTVlMS00M2NhLWE5Y2UtZTc3ZTkwYmUzYTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAxVDEwNDIzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjZjUwMWI4OTI5ODQzYzZhMTUyMGNmMTViNmUyMzJlOGE2NGVhYmZmN2U4ZmNhNDgzM2UxYTVlMzE5ZmNiYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.04Ci3jeJuza_vZFyntjaBJiGeylnQKaVwygozR4YnYc) Here we can see that the request is a `patch` request, resulting in a CORS error as the server returns `PATCH` as one of the allowed methods
> 
> The strange thing is, this seems to not be an issue in Safari, probably due to how the different browsers implement CORS checking.

You have to allow, cross origin in your server. It is not getx problem

**Comment by LittleSheep2Code** [comment]
> > Same issue here. Seems like the package is using `patch` instead of `PATCH`, which is why @Artel-Studio 's fix to `PATCH` works.
> > https://fetch.spec.whatwg.org/#methods https://stackoverflow.com/questions/73985866/the-mystery-of-patch-request-and-cors-headers
> > ![image](https://private-user-images.githubusercontent.com/54533009/327043944-7df81bbd-95e1-43ca-a9ce-e77e90be3a00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ1NjA0NTQsIm5iZiI6MTcxNDU2MDE1NCwicGF0aCI6Ii81NDUzMzAwOS8zMjcwNDM5NDQtN2RmODFiYmQtOTVlMS00M2NhLWE5Y2UtZTc3ZTkwYmUzYTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAxVDEwNDIzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjZjUwMWI4OTI5ODQzYzZhMTUyMGNmMTViNmUyMzJlOGE2NGVhYmZmN2U4ZmNhNDgzM2UxYTVlMzE5ZmNiYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.04Ci3jeJuza_vZFyntjaBJiGeylnQKaVwygozR4YnYc) Here we can see that the request is a `patch` request, resulting in a CORS error as the server returns `PATCH` as one of the allowed methods
> > The strange thing is, this seems to not be an issue in Safari, probably due to how the different browsers implement CORS checking.
> 
> You have to allow, cross origin in your server. It is not getx problem

No, this is getx's issue, the patch method needs to be all caps. I'm sure since my pure JavaScript code is working perfect with the same API and the same method, payload and headers.

https://github.com/JakeChampion/fetch/issues/254#issuecomment-169838138

<img width="677" alt="Pic1" src="https://github.com/user-attachments/assets/13b340c1-9706-4587-bae0-247ac15f26f3">

My code is also experienced the patch CORS issue, and I'm sure that I enabled CORS for every method (see the OPTION request result)

After I change the code from `GetConnect.patch` to `GetConnect.request('PATCH', ...)` it fixed.

<img width="765" alt="image" src="https://github.com/user-attachments/assets/17d1a0a9-3d34-4791-8be1-a855b68ef257">

Result: (the flutter we choose the chrome as dev browser, but the firefox part is also fixed)

<img width="652" alt="Pic3" src="https://github.com/user-attachments/assets/5536b28b-b651-457f-8c7c-60e58e989d27">

No more CORS issue.

**Comment by Cabanon** [comment]
I have the same issue, and this is actually a GetX problem, HTTP methods should be uppercase, and according to the HTTP spec 'patch' method do not exist and should be 'PATCH', the fix by @LittleSheep2Code works and should be added to the codebase

@jonataslaw Could you remove the "Unrelated to GetX" label so I can create a merge request ?
